### PR TITLE
feat(fork): Listen on a random port and send it to the parent process.

### PR DESCRIPTION
**Pull request by heathkit**
This will allow Protractor to fork instances of BlockingProxy as needed.

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -2,6 +2,7 @@ docs/
 lib/
 scripts/
 spec/
+testapp/
```
Comment (cnishina) [inline .npmignore:7]: should also have `built/spec` if your typescript files are being transpiled to `built`

File: lib/blockingproxy.ts [hunk 2]
```diff
@@ -111,7 +111,6 @@ export class BlockingProxy {
   };
 
   handleProxyCommand(message, data, response) {
-    console.log('Got message: ' + message.url);
     var command = message.url.split('/')[2];
```
Comment (cnishina) [inline lib/blockingproxy.ts]: several places in code that use `var` instead of preferred `let`

File: lib/blockingproxy.ts [hunk 3]
```diff
@@ -181,7 +180,6 @@ export class BlockingProxy {
                 reject(value);
                 return;
```
Comment (cnishina) [inline lib/blockingproxy.ts]: if you are rejecting the value, is the return required?

File: lib/config.ts [hunk 4]
```diff
@@ -4,21 +4,22 @@ import * as minimist from 'minimist';
 
 export interface Config {
   help?: boolean
+  fork?: boolean
```
Comment (cnishina) [inline lib/config.ts]: properties should end with a semi colon.

**Review by cnishina — CHANGES_REQUESTED** [review]

File: lib/blockingproxy.ts (hunk 2 above)
Comment (heathkit) [reply lib/blockingproxy.ts]: Yeah, I never really took care of that when I ported to typescript. Let me clean all that up in a separate PR.

Comment (cnishina) [reply lib/blockingproxy.ts]: SGTM.

**Review by cnishina — APPROVED** [review]

**Pull request merged**